### PR TITLE
v0.33 - Add pebble-based storage layer files

### PR DESCRIPTION
**Pull request by zhangchiqing**
This PR cherry-picked changes from https://github.com/onflow/flow-go/pull/6197. 

The changes picked are unused files or changes that won't change the existing behavior. Merging these changes in order to minimize the code changes to be maintained on a feature branch.

### CODE REVIEW

**Comment by zhangchiqing** [comment]
Just passed benchnet testing with 20tps benchmarking.

File: storage/pebble/operation/common.go [hunk 1]
```diff
@@ -34,17 +93,331 @@ func retrieve(key []byte, sc interface{}) func(r pebble.Reader) error {
...
+			if errors.Is(err, pebble.ErrNotFound) {
+				*keyExists = false
+				return nil
+			}
+
+			// exception while checking for the key
+			return irrecoverable.NewExceptionf("could not load data: %w", err)
+		}
+		*keyExists = true
+		defer closer.Close()
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply storage/pebble/operation/common.go:117]: I was curious so I benchmarked it: https://gist.github.com/jordanschalm/d4d268a39c6b2197a0c0882a4ca4dab5.

It seems like it has improved a lot in the past few years. About 5 years ago, it was 15x slower (~50ns) in a minimal benchmark. Now it's about 2x slower (2ns). 

Agree with removing it in this case, but I think the performance overhead is negligible now.

File: storage/pebble/procedure/cluster.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		// insert constituent transactions
+		for _, colTx := range payload.Collection.Transactions {
+			err = operation.InsertTransaction(colTx.ID(), colTx)(tx)
+			if err != nil {
+				return fmt.Errorf("could not insert payload transaction: %w", err)
+			}
+		}
+
+		// index the transaction IDs within the collection
+		txIDs := payload.Collection.Light().Transactions
```
Comment (AlexHentschel) [inline storage/pebble/procedure/cluster.go:179]: we already computed `payload.Collection.Light()` above. As this has a non-negligible computational cost, we should not repeat it here,

File: storage/pebble/procedure/cluster.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+			return fmt.Errorf("could not insert payload collection: %w", err)
+		}
+
+		// insert constituent transactions
+		for _, colTx := range payload.Collection.Transactions {
+			err = operation.InsertTransaction(colTx.ID(), colTx)(tx)
+			if err != nil {
+				return fmt.Errorf("could not insert payload transaction: %w", err)
+			}
+		}
```
Comment (AlexHentschel) [inline storage/pebble/procedure/cluster.go:176]: note that `payload.Collection.Light()` already computes the IDs of the transactions. This has non-negligible cost due to serialization and potentially hashing somewhat larger data structures. 

Lets not needlessly repeat that here. The `colTx.ID()` in line 172 is already pre-computed in `light`

**Comment by zhangchiqing** [comment]
Close in favor of https://github.com/onflow/flow-go/pull/6197

**Pull request closed without merging**